extensions/BOP/twitch.js

```diff
@@ -0,0 +1,324 @@
+            blockType: Scratch.BlockType.BOOLEAN,
+          },
+          {
+            opcode: "false",
+            text: "false",
+            blockType: Scratch.BlockType.BOOLEAN,
+          },
+        ],
+        menus: {
+          font: {
+            acceptReporters: false,
+            items: "getFonts",
+          },
+          position: {
+            acceptReporters: false,
+            items: "getPositions",
+          },
+        },
+      };
+    }
+
+    setupchat(args) {
+      if (chat == null) {
+        var Link =
+          "https://nightdev.com/hosted/obschat/?theme=bttv_dark&channel=" +
+          Scratch.Cast.toString(args.username) +
+          "&fade=" +
+          Scratch.Cast.toNumber(args.fade) +
+          "&bot_activity=" +
+          Scratch.Cast.toString(args.bots) +
+          "&prevent_clipping=" +
+          Scratch.Cast.toString(args.prevent);
+        chat = document.createElement("iframe");
+        chat.style.width = args.width;
+        chat.style.height = args.height;
+        chat.style.border = "none";
+        chat.style.position = args.position;
+        chat.setAttribute("allowtransparency", "true");
+        chat.setAttribute("src", Link);
+        Scratch.renderer.addOverlay(chat, "manual");
+      } else {
+        alert("Use close chat to close the chat.");
+      }
+    }
+
+    closechat() {
+      if (chat == null) {
+        alert("Chat hasn't been created yet!");
+      } else {
+        Scratch.renderer.removeOverlay(chat);
+        chat = null;
+      }
+    }
+
+    text(args) {
+      if (text == null) {
+        CreateTextElement(args);
+        Scratch.renderer.addOverlay(text, "manual");
+      } else {
+        Scratch.renderer.removeOverlay(text);
+        CreateTextElement(args);
+        Scratch.renderer.addOverlay(text, "manual");
+      }
+    }
+
+    image(args) {
+      if (image == null) {
+        CreateImageElement(args);
+        Scratch.renderer.addOverlay(image, "manual");
+      } else {
+        Scratch.renderer.removeOverlay(image);
+        CreateImageElement(args);
+        Scratch.renderer.addOverlay(image, "manual");
+      }
+    }
+
+    closetext() {
+      if (text == null) {
+        alert("No text element was found!");
+      } else {
+        Scratch.renderer.removeOverlay(text);
+        text = null;
+      }
+    }
+
+    closeimage() {
+      if (image == null) {
+        alert("No image element was found!");
+      } else {
+        Scratch.renderer.removeOverlay(image);
+        image = null;
+      }
+    }
+
+    true() {
+      return true;
+    }
+
+    false() {
+      return false;
+    }
+
+    // credits to lily: (animated text)
+
+    getFonts() {
+      const customFonts = Scratch.vm.runtime.fontManager
+        ? Scratch.vm.runtime.fontManager.getFonts().map((i) => ({
+            text: i.name,
+            value: i.family,
+          }))
+        : [];
+
+      return [
+        ...FONTS,
+        ...customFonts,
+        {
```

Comment:
Remove this part of the JSON object: you can't select a random font in this extension

Reply:
I did, but then I had to undo everything because of an error, so I will.

Reply:
✅